### PR TITLE
Implement backing beats for Nightcore mods

### LINKED ISSUE #6060: Missing beats in nightcore mod
**Describe the missing feature:** 

The beat sounds in nightcore mod are missing.

Stable: https://streamable.com/wlw87
Lazer; https://streamable.com/2c7io

### PR DESCRIPTION
**Pull request by LeNitrous**
Closes #6060 and requires https://github.com/ppy/osu-resources/pull/81.

Adds back the backing beats in the Nightcore mod.

### CODE REVIEW

**Comment by peppy** [comment]
Might as well add back the samples over at [osu-resources](https://github.com/ppy/osu-resources) and get the default working too.

File: osu.Game/Rulesets/Mods/ModNightcore.cs [hunk 1]
```diff
@@ -34,5 +43,47 @@ public override void ApplyToTrack(Track track)
...
+
+                if (beatIndex > -1)
+                {
+                    if (beatIndex % 16 == 0)
+                        finishSample?.Play();
+                    else if (beatIndex % 2 == 0)
+                        kickSample?.Play();
+                    else if (beatIndex % 2 == 1)
+                        clapSample?.Play();
+                    else
```
Comment (huoyaoyuan) [inline osu.Game/Rulesets/Mods/ModNightcore.cs]: How can this be reachable after `%2 == 0` and `%2 == 1`?

Comment (LeNitrous) [reply osu.Game/Rulesets/Mods/ModNightcore.cs]: I've based off the logic in smoogipoo's reply here https://github.com/ppy/osu/issues/6060#issuecomment-548198973 however I never saw the hat being used even in stable so I'm not quite sure about it.

**Comment by peppy** [comment]
This required further testing and comparison to stable:

- Didn't support simple triple timing
- Didn't play the correct beat (incorrect speed and also beat)
- Had unreachable code
- No test scene
- Continued to play the beat during pause

I've fixed all of these. @smoogipoo requesting re-review as I touched every line.

**Review by peppy — DISMISSED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**